ddtrace/internal/telemetry/writer.py

```diff
@@ -245,15 +260,11 @@ def enable(self):
         Enable the instrumentation telemetry collection service. If the service has already been
         activated before, this method does nothing. Use ``disable`` to turn off the telemetry collection service.
         """
-        with self._lock:
-            if self.status == ServiceStatus.RUNNING:
-                return
-
-            self.start()
-            self._enabled = True
+        if self.status == ServiceStatus.RUNNING:
+            return
 
-            forksafe.register(self._restart)
-            atexit.register(self.stop)
+        self.start()
+        self._enabled = True
 
-        # add_event _locks around adding to the events queue
-        self.app_started_event()
+        forksafe.register(self._fork_writer)
+        atexit.register(self.stop)
```

Comment:
minor refactor:
- I removed an unnecessary lock from `TelemetryWriter.enable()`
- moved `TelemetryWriter.app_started_event()` to `TelemetryWriter._start_service`. This event should be sent when the worker thread starts (this should not change the existing behavior, I think this better aligns with how we use PeriodicServices). 

I can move these changes to a seperate PR if that's preferred.

Reply:
> This event should be sent when the worker thread starts instead of being called in enable()

So the idea is to create/queue the event when the thread is created, but we won't actually try to flush it until it is enabled, right?

Reply:
Why is the lock no longer necessary?

Reply:
We can create/queue events after the global TelemetryWriter is initialized. We only flush/send events after the thread is created (via `TelemetryWriter.start`) and the writer is enabled.

The locks on enable/disable were never necessary. We only need locks around flushing and adding to the events queue